addons/website_slides/models/slide_channel.py

```diff
@@ -570,40 +649,81 @@ def action_channel_invite(self):
             'res_model': 'slide.channel.invite',
             'target': 'new',
             'context': local_context,
+            'name': _('Share a Course')
         }
 
     def action_add_member(self, **member_values):
-        """ Adds the logged in user in the channel members.
-        (see '_action_add_members' for more info)
-
-        Returns True if added successfully, False otherwise."""
-        return bool(self._action_add_members(self.env.user.partner_id, **member_values))
-
-    def _action_add_members(self, target_partners, **member_values):
-        """ Add the target_partner as a member of the channel (to its slide.channel.partner).
-        This will make the content (slides) of the channel available to that partner.
+        """ Enroll the logged in user in the course. (see '_action_add_members' for more info)
+
+        If the logged in user has been invited, then we sudo the add_members method
+        since we consider he has been granted access to the course.
+
+        Returns True if added and enrolled successfully, False otherwise."""
+        added_to_invited_course = False
+        courses_invited = self.filtered(lambda course: course.is_member_invitation_pending)
```

Comment:
Ok, a bit complicated to read but I understand. Purpose is the "/enroll" route, isn't it, that calls action_add_member I suppose ? A the private methods is mainly used in other cases. Would be worth a note :) .